src/com/goide/inspections/GoPlaceholderCountInspection.java

```diff
@@ -0,0 +1,209 @@
+        List<GoExpression> arguments = o.getArgumentList().getExpressionList();
+        if (arguments.size() <= placeholderPosition) {
+          return;
+        }
+
+        int parameterCount = arguments.size() - placeholderPosition - 1;
+        int placeholdersCount = getPlaceholders(arguments.get(placeholderPosition));
+        if (placeholdersCount == -1) {
+          return;
+        }
+
+        if (placeholdersCount != parameterCount) {
+          String message = String.format("Got %d placeholder(s) for %d arguments(s)", placeholdersCount, parameterCount);
+
+          PsiElement element = o.getExpression();
+          if (placeholdersCount < parameterCount) {
+            element = arguments.get(placeholderPosition);
+          }
+
+          holder.registerProblem(element, message, ProblemHighlightType.GENERIC_ERROR_OR_WARNING);
+        }
+      }
+    };
+  }
+
+  private static boolean isInCorrectPackage(@NotNull GoFunctionOrMethodDeclaration element) {
+    GoFile containingFile = element.getContainingFile();
+
+    return FORMATTING_FUNCTIONS.containsKey(getName(element)) &&
+           ("fmt".equals(containingFile.getImportPath()) ||
+            "log".equals(containingFile.getImportPath()));
+  }
+
+  @Nullable
+  private static String getName(@NotNull GoFunctionOrMethodDeclaration element) {
+    String name = element.getName();
+    if (name == null) {
+      return null;
+    }
+
+    return name.toLowerCase(Locale.getDefault());
+  }
+
+  private static int getPlaceholders(@Nullable GoExpression argument) {
+    if (argument == null) {
+      return -1;
+    }
+
+    String placeholderText = resolve(argument);
+    if (placeholderText == null) {
+      return -1;
+    }
+
+
+    Matcher placeholders = PLACEHOLDER_PATTERN.matcher(placeholderText);
+    Matcher countPlaceholders;
+
+    int indexCount = 0;
+    int maxIndexCount = 0;
+
+    while (placeholders.find()) {
+      indexCount++;
+
+      countPlaceholders = INDEXED_PLACEHOLDER_PATTERN.matcher(placeholders.group());
+      while (countPlaceholders.find()) {
+        int index = StringUtil.parseInt(countPlaceholders.group().replaceAll("\\[|\\]", ""), -1);
+        if (index > 0 && index != indexCount) {
+          indexCount = index;
+        }
+        if (maxIndexCount < indexCount) {
+          maxIndexCount = indexCount;
+        }
+      }
+      if (maxIndexCount < indexCount) {
+        maxIndexCount = indexCount;
+      }
+    }
+
+    return maxIndexCount;
+  }
+
+  @Nullable
+  private static String resolve(GoExpression argument) {
+    if (argument instanceof GoStringLiteral) {
+      return argument.getText();
+    }
+
+    if (!(argument instanceof GoReferenceExpression)) {
+      return null;
+    }
+
+    PsiReference resolved = argument.getReference();
+    if (resolved == null) {
+      return null;
+    }
+
+    PsiElement element = resolved.resolve();
+    while (element != null &&
```

Comment:
I don't understand what it stands for. Any test example? It doesn't covered at the moment: https://teamcity.jetbrains.com/viewLog.html?buildId=638588&buildTypeId=IntellijIdeaPlugins_Go_Test&tab=coverage_idea